src/osdc/ObjectCacher.cc

```diff
@@ -410,10 +410,13 @@ void ObjectCacher::Object::audit_buffers()
       ceph_assert(it->first == it->second->start());
     }
     if (it->first < offset) {
-      lderr(oc->cct) << "AUDIT FAILURE: " << it->first << " " << *it->second
-		     << " overlaps with previous bh " << *((--it)->second)
+      auto cur = it;
+      auto prev = std::prev(cur);
+      lderr(oc->cct) << "AUDIT FAILURE: " << it->first << " " << *cur->second
```

Comment:
if you intend to use `cur` for readability, could replace `it` with `cur` as well.